### ISSUE TITLE
The procedure to flash a firmware in USB Storage Boot Mode is not clear 

### ISSUE THREAD

**Issue by severin-lemaignan** [entropia/tip-toi-reveng#144]
*this issue tracks the experiments to flash anew a firmware onto the tiptoi in [USB Storage Boot Mode](https://github.com/entropia/tip-toi-reveng/wiki/Firmware#usb-storage-boot-mode-firmware-flashing)*

After an interrupted firmware update (reported in #143), the tiptoi pen looks completely bricked: it does not turn on/off, the [test mode](https://github.com/entropia/tip-toi-reveng/wiki/Firmware#test-mode) does not work, it is not recognized when plugged over USB.

However, I was able to start in [USB Storage Boot Mode](https://github.com/entropia/tip-toi-reveng/wiki/Firmware#usb-storage-boot-mode-firmware-flashing).

So, once in USB Storage Boot Mode, I've fired up my Windows 10 virtual machine, with Anyka's [BurnTool 3.0.18.2](https://www.dropbox.com/sh/ywiryex4q0bnteq/AACICyAFXUqGsS4S84b2CQuva?dl=0). BurnTool happily imports the firmware (`Update3202MT_fr.upd` in my case).

What to do next is less clear. I've ticked both Update mode and UDisk Burn and then clicked on Download. It apparently did something as after ~20 sec, the tiptoi eventually disconnected. But it did not reboot either...

What experiences do you have with this process?

**Comment by thawn** [comment]
for convenience, i am copying [Ulis comment in the other thread](https://github.com/entropia/tip-toi-reveng/issues/143#issuecomment-279265398) here:

> That's the problem: unfortunately noone has managed yet to flash a
tiptoi with burntool. Burntool creates a logfil, maybe the errors in
there can help.

**Comment by wernight** [comment]
https://maehw.wordpress.com/tag/tiptoi/ explains some may help.

**Comment by clean31** [comment]
> What to do next is less clear. I've ticked both Update mode and UDisk Burn and then clicked on Download. It apparently did something as after ~20 sec, the tiptoi eventually disconnected. But it did not reboot either...
> 
> What experiences do you have with this process?

OK, maybe it's little bit too late.

But, after importing the upd file and check the update and the udisk options, you must connect the tiptoi with usb then hold the power button and maintain the resistor between GPIO13 and GND during all process.

**Comment by uli42** [comment]
On Sun, Oct 13, 2019 at 2:19 PM clean31 <notifications@github.com> wrote:
> What to do next is less clear. I've ticked both Update mode and UDisk Burn and then clicked on Download. It apparently did something as after ~20 sec, the tiptoi eventually disconnected. But it did not reboot either...
>
> What experiences do you have with this process?
>
> OK, maybe it's little bit too late.
>
> But, after importing the upd file and check the update and the udisk options, you must connect the tiptoi with usb then hold the power button and maintain the resistor between GPIO13 and GND during all process.

Have you successfully flashed a pen with this method?

There are different types of TipTois around as mentioned here:
https://github.com/entropia/tip-toi-reveng/wiki/Firmware#usb-storage-boot-mode-firmware-flashing.
Could you please extend that wiki page with your finding(s)?

Thanks,

Uli

**Comment by clean31** [comment]
Yes
I had corrupted my pen during a software update with used batteries.
My pen is a 4GB B6.

With the method above and the software Burntool.exe v3.18 and the file Update3202MT_fr.upd, i have succesfully flashed my pen.
Thanks to all for the informations and the wiki.

I have updated the wiki page.

**Comment by uli42** [comment]
Thank you! We should encourage people with a broken pen to try that!

Uli

clean31 <notifications@github.com> schrieb am So., 13. Okt. 2019, 15:25:

**Comment by B-G-T** [comment]
I can confirm that it works fine!

**Comment by B-G-T** [comment]
One more remark: The 1 kOhm resistor can be removed after the pen has started. It is not necessary to keep it in place during the flash process.

**Comment by clean31** [comment]
In my case, it does not work if I remove the resistance or if I release the
power button ... I do not understand why, it stops the flash process.

**Comment by uli42** [comment]
Might be depending on the pen version. I am still wondering why we did not find this out much earlier. We already had it...

**Comment by B-G-T** [comment]
The level of port GPIO13 is checked only at the beginning of the SOC booting process. Hence the SOC remains in the selected mode until reset.
The power button must be kept pressed because the "auto-hold function" (incl. automatic power down) is active only in "user mode".
I experienced also failed flash processes but from what I could see it's not related to disconnecting the resistor.

**Comment by B-G-T** [comment]
Flashing is now understood and works.
Can we close this issue?

**Comment by maehw** [comment]
Sounds great.

Three questions:

1. Has anyone further tried to sniff the USB data with a packet analyzer?
2. Has anyone recorder further UART debug output on  GPIO13? (We've seen a debug message here earlier.)
3. Has anyone tried the update mechanism under Linux with a Windows VM?

Cheers

**Comment by maehw** [comment]
Update:

I cannot reproduce flashing a pen with the BurnTool.
Is there anything specific one should do? Can somebody provide a detailed step-by-step guide, please?

Could anyone provide a detailed log of the USB data (this can be done wit Wireshark + USBPcap), this would be very much appreciated.
My findings concerning the SCSI-based USB mass storage boot are currently documented here: https://github.com/maehw/snowbirdopter/wiki/USB

**Comment by B-G-T** [comment]
Can you enter the Storage Boot Mode (https://github.com/entropia/tip-toi-reveng/wiki/PEN-Firmware)?

**Comment by maehw** [comment]
Yes, I've even identified Boot ROM code responsible for the SCSI commands that are used, but it's not 100% clear. See also: https://github.com/maehw/snowbirdopter/wiki/USB

**Comment by nomeata** [comment]
Is this discussion still active or can I close this issue?

**Comment by jurgenvdven** [comment]
Hi all, 

I tried to flash my tiptoi B6 using method 1
https://github.com/entropia/tip-toi-reveng/wiki/PEN-Firmware

But get stuck with step 5, when I try to get the device into boot mode I hear the Chime sound but without talking. 
So it seems that the resistor is doing something. (because without resistor i hear a chime sound and talking)

Unclear if this is enough, when i connect the device with USB, the USB drive is mapped and the ankay burn tool doesn't start automatically to download, When i click on download it is doing something like disconnect usb and connect again but the USB drive is mapped again.

I tried different scenario's but without a solution.

What could be the problem?

**Comment by B-G-T** [comment]
Hi,

As you hear the chime sound, I suppose you haven't entered the Storage Boot Mode.

Do you have the possibility to read the log output which is sent on GPIO13?

**Comment by jurgenvdven** [comment]
How can i read from gpio 13?

I have an usb uart device but not sure how to connect on tiptoi

**Comment by B-G-T** [comment]
You connect GPIO13 (Tiptoi UART TXD) to the UART RXD pin of your USB UART device. Of course, GND needs to be connected too.
Then you listen at 38400 bps.

That's all.

**Comment by jurgenvdven** [comment]
thank you for your comment. The tiptoi is working.

When de tiptoi is booting in normal mode, is say's "Hello, please wait" so i expect that when the pen is ready there will be another sound  or voice but it does not say anything or sound. Also the manual from ravensburger notes when the pen led  is flashing a reset is needed. But that was not the case. 

After all i found that when the pen says "hello, please wait" open a book  and hit something and the pen is working as expected. After a normal reboot the pen say's "Hello" without please wait.

I don't know human error or crappy instructions.
